### PR TITLE
Fix rendering of descriptor properties in docs

### PR DESCRIPTION
**Pull request by rlnx**
- Render descriptor properties correctly
- Fix cross references across API Reference
- Fix some critical issues with doxygen comments in sources which lead to build error and warnings
- Automatically resolve `v1` namespace in `dalapi` tool:
  - `v1` is not rendered in docs and no longer required to specify in RST
  - `v1` is removed from all RST documents
  - `NameTransformer` entity is introduced in `dalapi/loader.py` for fully qualified name transformation
- Remove `detail` namespace mentions from API reference

<h3><a href="https://user-images.githubusercontent.com/5404873/112156304-2e020780-8bf7-11eb-9139-1f9e8d7244fa.png">SVM docs screenshot<p/></h3>

### CODE REVIEW

File: cpp/oneapi/dal/algo/svm/common.hpp [hunk 1]
```diff
@@ -250,42 +218,100 @@ class descriptor : public detail::descriptor_base<Task> {
...
+        return base_t::get_shrinking();
+    }
+
     auto& set_shrinking(bool value) {
         base_t::set_shrinking_impl(value);
         return *this;
     }
 
+    template <typename T = Task, typename = detail::enable_if_classification_t<T>>
+    /// The class count. Used with :expr:`task::classification` only.
```
Comment (outoftardis) [inline cpp/oneapi/dal/algo/svm/common.hpp:294]:
```suggestion
    /// The number of classes. Used with :expr:`task::classification` only.
```

File: cpp/oneapi/dal/algo/svm/common.hpp [hunk 2]
```diff
@@ -250,42 +218,100 @@ class descriptor : public detail::descriptor_base<Task> {
...
+    double get_cache_size() const {
+        return base_t::get_cache_size();
+    }
+
     auto& set_cache_size(double value) {
         base_t::set_cache_size_impl(value);
         return *this;
     }
 
+    /// The parameter of the WSS scheme $\\tau$.
```
Comment (outoftardis) [inline cpp/oneapi/dal/algo/svm/common.hpp:269]: what is WSS?

File: cpp/oneapi/dal/algo/svm/common.hpp [hunk 3]
```diff
@@ -250,42 +218,100 @@ class descriptor : public detail::descriptor_base<Task> {
...
+    double get_accuracy_threshold() const {
+        return base_t::get_accuracy_threshold();
+    }
+
+    auto& set_accuracy_threshold(double value) {
+        base_t::set_accuracy_threshold_impl(value);
+        return *this;
+    }
+
+    /// The size of cache in megabytes for storing values of the kernel matrix.
```
Comment (outoftardis) [inline cpp/oneapi/dal/algo/svm/common.hpp:257]:
```suggestion
    /// The size of cache (in megabytes) for storing the values of the kernel matrix.
```

File: cpp/oneapi/dal/algo/jaccard/common.hpp [hunk 4]
```diff
@@ -67,17 +59,42 @@ class ONEDAL_EXPORT descriptor_base : public base {
...
+    std::int64_t get_row_range_begin() const {
+        return base_t::get_row_range_begin();
+    }
+
+    /// Returns the end of the row of the graph block
+    std::int64_t get_row_range_end() const {
+        return base_t::get_row_range_end();
+    }
+
+    /// Returns the begin of the column of the graph block
```
Comment (outoftardis) [inline cpp/oneapi/dal/algo/jaccard/common.hpp:82]:
```suggestion
    /// Returns the beginning of the column of the graph block
```

File: cpp/oneapi/dal/algo/jaccard/common.hpp [hunk 5]
```diff
@@ -67,17 +59,42 @@ class ONEDAL_EXPORT descriptor_base : public base {
...
 class descriptor : public descriptor_base {
+    using base_t = descriptor_base;
+
 public:
     using method_t = Method;
 
+    /// Creates a new instance of the class with the default property values.
+    descriptor() = default;
+
+    /// Returns the begin of the row of the graph block
```
Comment (outoftardis) [inline cpp/oneapi/dal/algo/jaccard/common.hpp:72]:
```suggestion
    /// Returns the beginning of the row of the graph block
```

**Review by outoftardis — APPROVED** [review]
I don't have any other comments
LGTM

File: cpp/oneapi/dal/algo/svm/common.hpp (hunk 2 above)
Comment (rlnx) [reply cpp/oneapi/dal/algo/svm/common.hpp:269]: @PetrovKP, probably we should not refer to WSS in API reference, the meaning is not clear from the context. Suggest better description for this param.

Comment (PetrovKP) [reply cpp/oneapi/dal/algo/svm/common.hpp:269]: I agree. I will fix

**Pull request merged**